### ISSUE TITLE
Query: perform null protection pattern (a != null ? a.Name : null) to NullConditionalExpression (a?.Name) translation on QueryModel level

### ISSUE THREAD

**Issue by maumar** [aspnet/EntityFramework#8385]
Currently we only perform this translation when running SqlTranslatingExpressionVisitor. However, we could apply some of them on the query model level, which would result in simpler queries, especially around such patterns on entities with composite keys.

**Comment by maumar** [comment]
This also applies to relational: Complex navigation test GroupJoin_on_complex_subquery_and_set_operation_on_grouping_but_nothing_from_grouping_is_projected":

```cs
            AssertQuery<Level1>(
                  l1s =>
                      l1s.GroupJoin(
                          l1s.Where(l1 => l1.Name != "L1 01").Select(l1 => l1.OneToOne_Required_FK),
                          l1 => l1.Id,
                          l2 => l2 != null ? l2.Level1_Optional_Id : null,
                          (l1, l2s) => new { l1, l2s })
                      .Where(r => r.l2s.Any())
                      .Select(r => r.l1),
                  e => e.Id,
                  (e, a) => Assert.Equal(e.Id, a.Id));
```

produces the following sql:

```sql
SELECT [l1].[Id], [l1].[Date], [l1].[Name], [l1].[OneToMany_Optional_Self_InverseId], [l1].[OneToMany_Required_Self_InverseId], [l1].[OneToOne_Optional_SelfId], [t].[Id], [t].[Date], [t].[Level1_Optional_Id], [t].[Level1_Required_Id], [t].[Name], [t].[OneToMany_Optional_InverseId], [t].[OneToMany_Optional_Self_InverseId], [t].[OneToMany_Required_InverseId], [t].[OneToMany_Required_Self_InverseId], [t].[OneToOne_Optional_PK_InverseId], [t].[OneToOne_Optional_SelfId]
FROM [Level1] AS [l1]
LEFT JOIN (
    SELECT [l1.OneToOne_Required_FK].[Id], [l1.OneToOne_Required_FK].[Date], [l1.OneToOne_Required_FK].[Level1_Optional_Id], [l1.OneToOne_Required_FK].[Level1_Required_Id], [l1.OneToOne_Required_FK].[Name], [l1.OneToOne_Required_FK].[OneToMany_Optional_InverseId], [l1.OneToOne_Required_FK].[OneToMany_Optional_Self_InverseId], [l1.OneToOne_Required_FK].[OneToMany_Required_InverseId], [l1.OneToOne_Required_FK].[OneToMany_Required_Self_InverseId], [l1.OneToOne_Required_FK].[OneToOne_Optional_PK_InverseId], [l1.OneToOne_Required_FK].[OneToOne_Optional_SelfId]
    FROM [Level1] AS [l10]
    LEFT JOIN [Level2] AS [l1.OneToOne_Required_FK] ON [l10].[Id] = [l1.OneToOne_Required_FK].[Level1_Required_Id]
    WHERE ([l10].[Name] <> N'L1 01') OR [l10].[Name] IS NULL
) AS [t] ON [l1].[Id] = CASE
    WHEN [t].[Id] IS NOT NULL
    THEN [t].[Level1_Optional_Id] ELSE NULL
END
ORDER BY [l1].[Id]
```

the CASE statement in the inner key selector for the first OUTER JOIN is not needed, we could just access t.Level1_Optional_Id directly

**Comment by maumar** [comment]
prerequisite for some issues related to null propagation, e.g. #6618 and #7761

**Comment by maumar** [comment]
fixed in 1bc0fe3b0cabc2ba38762d496114fdf42daf6fd3
